bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeHomeHandler.java

```diff
@@ -97,96 +100,164 @@ protected NAWelcomeHome updateReadings() {
             case CHANNEL_WELCOME_EVENT_SNAPSHOT_URL:
-                if (lastEvent != null) {
-                    String snapshotURL = getSnapshotURL(lastEvent.getSnapshot());
-                    return toStringType(snapshotURL);
-                } else {
-                    return UnDefType.UNDEF;
-                }
+                return findSnapshotURL().map(ChannelTypeUtils::toStringType).orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_EVENT_VIDEO_URL:
-                if (lastEvent != null && lastEvent.getVideoId() != null) {
-                    String cameraId = lastEvent.getCameraId();
+                if (lastEvent.isPresent() && lastEvent.get().getVideoId() != null) {
+                    String cameraId = lastEvent.get().getCameraId();
                     Optional<AbstractNetatmoThingHandler> thing = getBridgeHandler().findNAThing(cameraId);
                     if (thing.isPresent()) {
                         NAWelcomeCameraHandler eventCamera = (NAWelcomeCameraHandler) thing.get();
-                        String streamUrl = eventCamera.getStreamURL(lastEvent.getVideoId());
+                        String streamUrl = eventCamera.getStreamURL(lastEvent.get().getVideoId());
                         if (streamUrl != null) {
                             return new StringType(streamUrl);
                         }
                     }
                 }
                 return UnDefType.UNDEF;
             case CHANNEL_WELCOME_EVENT_VIDEOSTATUS:
-                return lastEvent != null ? toStringType(lastEvent.getVideoStatus()) : UnDefType.UNDEF;
+                return lastEvent.map(e -> toStringType(e.getVideoStatus())).orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_EVENT_ISARRIVAL:
-                return lastEvent != null ? lastEvent.getIsArrival() != null ? OnOffType.ON : OnOffType.OFF
-                        : UnDefType.UNDEF;
+                return lastEvent.map(e -> toOnOffType(e.getIsArrival())).orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_EVENT_MESSAGE:
-                return lastEvent != null && lastEvent.getMessage() != null
-                        ? new StringType(lastEvent.getMessage().replace("<b>", "").replace("</b>", ""))
-                        : UnDefType.UNDEF;
+                return findEventMessage().map(
+                        m -> (State)new StringType(m.replace("<b>", "").replace("</b>", ""))
+                ).orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_EVENT_SUBTYPE:
-                return lastEvent != null ? toDecimalType(lastEvent.getSubType()) : UnDefType.UNDEF;
+                return lastEvent.map(e -> toDecimalType(e.getSubType())).orElse(UnDefType.UNDEF);
         }
         return super.getNAThingProperty(channelId);
     }
 
+    @Override
+    protected void triggerChannelIfRequired(String channelId) {
+        if(isNewLastEvent) {
+            if(CHANNEL_CAMERA_EVENT.equals(channelId)) {
```

Comment:
space between if and '(' (doesn't the IDE format this automatically ?)

Reply:
done